docs/content/guides/dagster-pipes/subprocess/reference.mdx

```diff
@@ -256,3 +256,76 @@ defs = Definitions(
 
 </TabItem>
 </TabGroup>
+
+---
+
+## Passing custom data
+
+Sometimes, you may want to pass data back from the external process for use in the orchestration code for purposes other than reporting directly to Dagster.
+
+<TabGroup>
+<TabItem name="External code in external_code.py">
+
+In the external code, we send messages using `report_custom_message`. The message can be any data that is JSON serializable.
+
+```python file=/guides/dagster/dagster_pipes/subprocess/custom_messages/external_code.py
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    data = {"type": "example_msg", "data": 42}  # send any JSON serializable data
```

Comment:
Can we use a real example here? Even if it's something like reporting how many records were processed - we get a fair number of complaints about overly simplistic examples.